libevmasm/Assembly.cpp

```diff
@@ -1258,10 +1259,10 @@ LinkerObject const& Assembly::assembleLegacy() const
 		if (item.type() == PushTag)
 		{
 			auto [subId, tagId] = item.splitForeignPushTag();
-			if (subId == std::numeric_limits<size_t>::max())
+			if (subId.empty())
 				continue;
-			assertThrow(subId < m_subs.size(), AssemblyException, "Invalid sub id");
-			auto subTagPosition = m_subs[subId]->m_tagPositionsInBytecode.at(tagId);
+			solRequire(subId.value < m_subs.size(), AssemblyException, "Invalid sub id");
```

Comment:
This looks more like like an assertion to me. I don't think there is any user input that we'd expect to trigger it.

```suggestion
			solAssert(subId.value < m_subs.size(), "Invalid sub id");
```

Same below, in `tagRefs` loop.

Reply:
Also, why remove them for `PushSub` and `PushSubSize` rather than turn into asserts? They seem still relevant and `SubAssemblyID` constructor will not reject an empty value.

Reply:
> Also, why remove them for `PushSub` and `PushSubSize` rather than turn into asserts? 

Honestly I don't recall. Perhaps I did have an assert in the constructor at some point or I figured that the assert can never be triggered. In any case, putting them back, it doesn't hurt to have them.

Reply:
Now I recall :)

```cpp
solAssert(item.data() <= std::numeric_limits<size_t>::max());
```

this will fail for emscripten when `size_t` is just 32 bits. We have to change the assert to

```cpp
solAssert(item.data() <= std::numeric_limits<SubAssemblyID::ValueType>::max());
```

for it to be semantically the same. But this is trivially true.